File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/UserRoleMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class UserRoleMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "userRoleIdUserId",
            "userRoleIdRoleId",
            "userVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private UserRoleMvoStateEventId value;

    public UserRoleMvoStateEventIdFlattenedDto()
    {
        this(new UserRoleMvoStateEventId());
    }

    public UserRoleMvoStateEventIdFlattenedDto(UserRoleMvoStateEventId value)
    {
        this.value = value;
    }

    public UserRoleMvoStateEventId toUserRoleMvoStateEventId()
    {
        return this.value;
    }


    public String getUserRoleIdUserId()
    {
        return this.value.getUserRoleId().getUserId();
    }

    public void setUserRoleIdUserId(String userRoleIdUserId)
    {
        this.value.getUserRoleId().setUserId(userRoleIdUserId);
    }

    public String getUserRoleIdRoleId()
    {
        return this.value.getUserRoleId().getRoleId();
    }

    public void setUserRoleIdRoleId(String userRoleIdRoleId)
    {
        this.value.getUserRoleId().setRoleId(userRoleIdRoleId);
    }

    public Long getUserVersion()
    {
        return this.value.getUserVersion();
    }

    public void setUserVersion(Long userVersion)
    {
        this.value.setUserVersion(userVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        UserRoleMvoStateEventIdFlattenedDto other = (UserRoleMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/UserRoleMvoStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class UserRoleMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'userRoleIdUserId' => 'string',
            'userRoleIdRoleId' => 'string',
            'userVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var UserRoleMvoStateEventId
     */
    private $value;

    /**
     * @param UserRoleMvoStateEventId $value
     */
    public function __construct(UserRoleMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new UserRoleMvoStateEventId();
        }
    }

    /**
     * @return UserRoleMvoStateEventId
     */
    public function toUserRoleMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getUserRoleIdUserId()
    {
        return $this->value->getUserRoleId()->getUserId();
    }

    /**
     * @param string $userRoleIdUserId
     */
    public function setUserRoleIdUserId($userRoleIdUserId)
    {
        $this->value->getUserRoleId()->setUserId($userRoleIdUserId);
    }

    /**
     * @return string
     */
    public function getUserRoleIdRoleId()
    {
        return $this->value->getUserRoleId()->getRoleId();
    }

    /**
     * @param string $userRoleIdRoleId
     */
    public function setUserRoleIdRoleId($userRoleIdRoleId)
    {
        $this->value->getUserRoleId()->setRoleId($userRoleIdRoleId);
    }

    /**
     * @return Long
     */
    public function getUserVersion()
    {
        return $this->value->getUserVersion();
    }

    /**
     * @param Long $userVersion
     */
    public function setUserVersion($userVersion)
    {
        $this->value->setUserVersion($userVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getUserRoleIdUserId(),
            $this->getUserRoleIdRoleId(),
            $this->getUserVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return UserRoleMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setUserRoleIdUserId($pValues[0]);
        $this->setUserRoleIdRoleId($pValues[1]);
        $this->setUserVersion($pValues[2]);
        return $this;
    }

}

